include/avif/avif.h

```diff
@@ -48,6 +48,21 @@ extern "C" {
 #define AVIF_API
 #endif // defined(AVIF_DLL)
 
+#if defined(AVIF_ENABLE_NODISCARD) || (defined(__cplusplus) && __cplusplus >= 201700L) || \
+    (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L)
+#define AVIF_NODISCARD [[nodiscard]]
+#else
+#if defined(__clang__) && defined(__has_attribute)
+#if __has_attribute(warn_unused_result)
+#define AVIF_NODISCARD __attribute__((warn_unused_result))
```

Comment:
A difference between `__attribute__((warn_unused_result))` and `[[nodiscard]]`, which I ran into, is that `__attribute__((warn_unused_result))` doesn't allow us to use a `void` cast to mark a function's result as used. In most cases this doesn't really matter.

Reply:
Thx, good to know !